### PR TITLE
how-to: document how to scale with TiUP

### PR DESCRIPTION
**Pull request by wyuchen007**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: how-to/scale/with-tiup.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+title: 使用 TiUP 扩容缩容 TiDB 集群
+category: how-to
+---
+
+# 使用 TiUP 扩容缩容 TiDB 集群
+
+TiDB 集群可以在不影响线上服务的情况下进行扩容和缩容。
+
+本文介绍如何使用 TiUP 扩容缩容集群中的 TiDB、TiKV 或者 PD 节点。
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:10]: 既然 TiFlash 已经加到 TiUP 的部署文档，那扩缩容文档要加上如何扩缩容 TiFlash 吗？ @superlzs0476 @ilovesoup 

或者提供个链接到 https://pingcap.com/docs-cn/dev/reference/tiflash/scale/？

File: how-to/scale/with-tiup.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+```
+
+预期输出 Scaled cluster `test` in successfully 信息，表示扩容操作成功。
+
+### 1.3 检查集群状态
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tiup cluster display testy
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:79]:
```suggestion
tiup cluster display test
```

File: how-to/scale/with-tiup.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+### 2.1 查看节点 ID 信息
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tiup cluster display test
+```
+
+```
+Starting /root/.tiup/components/cluster/v0.3.3/cluster display testy
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:101]:
```suggestion
Starting /root/.tiup/components/cluster/v0.3.3/cluster display test
```

File: how-to/scale/with-tiup.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+10.0.1.5:9293   alertmanager 10.0.1.5  9293/9294    Up      data/alertmanager-9293  deploy/alertmanager-9293
+```
+
+### 2.2 执行缩容操作
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tiup cluster scale-in testy --node 10.0.1.5:20160
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:135]:
```suggestion
tiup cluster scale-in test --node 10.0.1.5:20160
```

File: how-to/scale/with-tiup.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+### 2.3 检查集群状态
+
+下线需要一定时间，下线节点的状态变为 Tombstone 就说明下线成功。
+
+执行如下命令检查节点是否下线成功：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tiup cluster display testy
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:151]:
```suggestion
tiup cluster display test
```

File: how-to/scale/with-tiup.md (hunk 1 above)
Comment (yikeke) [reply how-to/scale/with-tiup.md:10]: 已和 @superlzs0476 @ilovesoup 确认，需要将 https://pingcap.com/docs-cn/dev/reference/tiflash/scale/ 的内容整合到 how-to/scale/with-tiup.md 文件里。

File: how-to/scale/with-tiup.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+> **注意：**
+>
+> 添加 TiKV 和 PD 节点和添加 TiDB 节点的步骤类似。
+
+### 1.1 编写扩容拓扑配置
+
+> **注意：**
+>
+> 端口默认可以不填，但是单机多实例需要分配不同的端口，如果有端口或目录冲突，会在部署或扩容时提醒。
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:33]:
```suggestion
> 端口默认可以不填，但是单机多实例需要分配不同的端口，如果有端口或目录冲突，会在部署或扩容时提醒。
```
```suggestion
> 默认情况下，可以不填端口。但在单机多实例场景下，你需要分配不同的端口，如果有端口或目录冲突，会在部署或扩容时提醒。
```

File: how-to/scale/with-tiup.md (hunk 1 above)
Comment (superlzs0476) [reply how-to/scale/with-tiup.md:10]: 确认

File: how-to/scale/with-tiup.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+>
+> 添加 TiKV 和 PD 节点和添加 TiDB 节点的步骤类似。
+
+### 1.1 编写扩容拓扑配置
+
+> **注意：**
+>
+> 默认情况下，可以不填端口信息。但在单机多实例场景下，你需要分配不同的端口，如果有端口或目录冲突，会在部署或扩容时提醒。
+
+在 scale-out.yaml 文件添加扩容拓扑配置：
```
Comment (kissmydb) [inline how-to/scale/with-tiup.md:35]:
```suggestion
新建 scale-out.yaml 文件，添加扩容拓扑配置：
```

File: how-to/scale/with-tiup.md [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+本文介绍如何使用 TiUP 扩容缩容集群中的 TiDB、TiKV、PD 或者 TiFlash 节点。
+
+原拓扑结构如下所示：
+
+| 主机 IP   | 服务   | 
+|:----|:----|
+| 10.0.1.4   | TiDB + PD    | 
+| 10.0.1.5   | TiKV + Monitor   | 
+| 10.0.1.1   | TiKV   | 
+| 10.0.1.2   | TiKV    |
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:19]: 因为加进来 TiFlash 的内容，这篇文档的几处的拓扑结构需要更新下吗？ @wyuchen007 @superlzs0476

File: how-to/scale/with-tiup.md (hunk 3 above)
Comment (yikeke) [reply how-to/scale/with-tiup.md:134]: 这里不用改吗 @superlzs0476

File: how-to/scale/with-tiup.md [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+| 10.0.1.4   | TiDB + PD    | 
+| 10.0.1.5   | TiDB + Monitor**（TiKV 已删除）**   | 
+| 10.0.1.1   | TiKV    | 
+| 10.0.1.2   | TiKV    | 
+
+打开浏览器访问监控平台 <http://10.0.1.5:3200>，监控整个集群和新增节点的状态。
+
+## 4. 缩容 TiFlash 节点
+
+如果要下线一个 TiFlash 节点，IP 地址为 172.19.0.104，可以按照如下步骤进行操作。
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:200]:
```suggestion
如果要集群中移除一个 TiFlash 节点，IP 地址为 172.19.0.104，可以按照如下步骤进行操作。
```

File: how-to/scale/with-tiup.md [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+
+> **注意：**
+>
+> 本节介绍的下线流程不会删除缩容节点上的数据文件，如需再次上线，请先手动删除。
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:204]:
```suggestion
> 本节介绍的缩容流程不会删除缩容节点上的数据文件，如需再次上线该节点，请先手动删除原来的数据文件。
```

File: how-to/scale/with-tiup.md [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+
+### 4.1 下线该 TiFlash 节点
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:206]:
```suggestion
### 4.1 下线 TiFlash 节点
```

File: how-to/scale/with-tiup.md (hunk 10 above)
Comment (yikeke) [reply how-to/scale/with-tiup.md:204]: @ilovesoup 这样改可以不

File: how-to/scale/with-tiup.md [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+
+ - host: 10.0.1.5
+
+   ssh_port: 22
+
+   port: 4000
+
+   status_port: 10080
+```
+
```
Comment (kissmydb) [inline how-to/scale/with-tiup.md:54]:
```suggestion
当前集群的整体配置，可以使用 `tiup cluster edit-config <cluster-name>` 查看，其中的 global 和 server_configs 的全局配置也会在 `scale-out.yaml` 中生效

```

**Comment by kissmydb** [comment]
rest LGTM

File: how-to/scale/with-tiup.md [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+   ssh_port: 22
+
+   port: 4000
+
+   status_port: 10080
+```
+你可以使用 `tiup cluster edit-config <cluster-name>` 查看当前集群的整体配置，其中 global 和 server_configs 的全局配置也会在 `scale-out.yaml` 中生效。
+
+当前集群的整体配置，可以使用 `tiup cluster edit-config <cluster-name>` 查看，其中的 global 和 server_configs 的全局配置也会在 `scale-out.yaml` 中生效
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:56]:
```suggestion
```

File: how-to/scale/with-tiup.md [hunk 14]
```diff
@@ -0,0 +1,235 @@
...
+
+ - host: 10.0.1.5
+
+   ssh_port: 22
+
+   port: 4000
+
+   status_port: 10080
+```
+你可以使用 `tiup cluster edit-config <cluster-name>` 查看当前集群的整体配置，其中 global 和 server_configs 的全局配置也会在 `scale-out.yaml` 中生效。
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:54]:
```suggestion

你可以使用 `tiup cluster edit-config <cluster-name>` 查看当前集群的整体配置，其中 global 和 server_configs 的全局配置也会在 `scale-out.yaml` 中生效。
```

File: how-to/scale/with-tiup.md [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+
+   ssh_port: 22
+
+   port: 4000
+
+   status_port: 10080
+```
+
+你可以使用 `tiup cluster edit-config <cluster-name>` 查看当前集群的整体配置，其中 global 和 server_configs 的全局配置也会在 `scale-out.yaml` 中生效。
+
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:56]:
```suggestion
```

File: how-to/scale/with-tiup.md [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+>
+> 添加 TiKV 和 PD 节点和添加 TiDB 节点的步骤类似。
+
+### 1.1 编写扩容拓扑配置
+
+> **注意：**
+>
+> 默认情况下，可以不填端口信息。但在单机多实例场景下，你需要分配不同的端口，如果有端口或目录冲突，会在部署或扩容时提醒。
+
+新建 scale-out.yaml 文件，添加扩容拓扑配置：
```
Comment (yikeke) [inline how-to/scale/with-tiup.md:35]:
```suggestion
在 scale-out.yaml 文件添加扩容拓扑配置：
```

File: how-to/scale/with-tiup.md [hunk 17]
```diff
@@ -195,38 +197,38 @@ tiup cluster display test
...
-如果要集群中移除一个 TiFlash 节点，IP 地址为 172.19.0.104，可以按照如下步骤进行操作。
+如果要下线一个 TiFlash 节点，IP 地址为 172.19.0.104，可以按照如下步骤进行操作。
 
 > **注意：**
 >
-> 本节介绍的缩容流程不会删除缩容节点上的数据文件，如需再次上线该节点，请先手动删除原来的数据文件。
+> 本节介绍的下线流程不会删除缩容节点上的数据文件，如需再次上线，请先手动删除。
 
-### 4.1 下线 TiFlash 节点
+### 3.1 下线该 TiFlash 节点
```
Comment (kissmydb) [inline how-to/scale/with-tiup.md:208]:
```suggestion
### 4.1 下线 TiFlash 节点
```

File: how-to/scale/with-tiup.md [hunk 18]
```diff
@@ -0,0 +1,245 @@
+---
```
Comment (ran-huang) [inline how-to/scale/with-tiup.md:1]: This doc is not listed in `TOC.md`

File: how-to/scale/with-tiup.md [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tiup cluster scale-in <cluster-name> --node 10.0.1.5:20160
+```
+
+其中 `--node` 参数为需要下线节点的 ID。
+
+预期输出 Scaled cluster `<cluster-name>` in successfully 信息，表示扩容操作成功。
```
Comment (ran-huang) [inline how-to/scale/with-tiup.md:185]:
```suggestion
预期输出 Scaled cluster `<cluster-name>` in successfully 信息，表示缩容操作成功。
```

File: how-to/scale/with-tiup.md [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+
+### 1.2 执行扩容命令
+
+{{< copyable "shell-regular" >}}
+
+```shell
+tiup cluster scale-out <cluster-name> scale-out.yaml
+```
+
+预期输出 Scaled cluster `<cluster-name>` in successfully 信息，表示扩容操作成功。
```
Comment (ran-huang) [inline how-to/scale/with-tiup.md:78]:
```suggestion
预期输出 Scaled cluster `<cluster-name>` out successfully 信息，表示扩容操作成功。
```

File: how-to/scale/with-tiup.md [hunk 21]
```diff
@@ -0,0 +1,245 @@
...
+
+| Host IP   | Service   | 
+|:----|:----|
+| 10.0.1.3   | TiDB + TiFlash   |
+| 10.0.1.4   | TiDB + PD + TiFlash   | 
+| 10.0.1.5   | TiDB + Monitor**（TiKV 已删除）**   | 
+| 10.0.1.1   | TiKV    | 
+| 10.0.1.2   | TiKV    | 
+
+打开浏览器访问监控平台 <http://10.0.1.5:3200>，监控整个集群和新增节点的状态。
```
Comment (ran-huang) [inline how-to/scale/with-tiup.md:209]: I think there is no `new node` after the scaling-in operation
```suggestion
打开浏览器访问监控平台 <http://10.0.1.5:3200>，监控整个集群的状态。
```

**Pull request merged**